example/src/main/java/io/netty/example/objectecho/ObjectEchoClientHandler.java

```diff
@@ -41,9 +42,10 @@ public ObjectEchoClientHandler() {
     }
 
     @Override
-    public void channelActive(ChannelHandlerContext ctx) {
+    public void channelActive(ChannelHandlerContext ctx) throws Exception {
         // Send the first message if this handler is a client-side handler.
-        ctx.writeAndFlush(firstMessage);
+        ChannelFuture future = ctx.writeAndFlush(firstMessage);
+        future.sync(); // Let object serialisation exceptions propagate.
```

Comment:
@chrisvest this doesn't work as this will block the eventloop. Use a `ChannelFutureListener`